### PR TITLE
Test for codestream lifecycle information being correct (bsc#1227100)

### PR DESCRIPTION
**Pull request by dirkmueller**
[CI:TOXENVS] repository

### CODE REVIEW

**Review by dcermak — COMMENTED** [review]
This still needs a reformat and also the repo currently shows 2028 as the EOL date, so I guess this needs to be fixed in SLES?

**Comment by dirkmueller** [comment]
> This still needs a reformat and also the repo currently shows 2028 as the EOL date, so I guess this needs to be fixed in SLES?

Yeah, I'm following TDD

**Comment by dcermak** [comment]
What is blocking getting this fixed on the SLES side? Something in the SLE_BCI repo that we can fix?

**Comment by dirkmueller** [comment]
> What is blocking getting this fixed on the SLES side? Something in the SLE_BCI repo that we can fix?

the information comes from the sles-release package. since it was wrong in SP6:GA, we need to release a maintenance update that noone has submitted so far.

File: tests/test_repository.py [hunk 1]
```diff
@@ -157,6 +157,31 @@ def test_package_installation(container_per_test, pkg):
     )
 
 
+@pytest.mark.skipif(
+    OS_VERSION not in ALLOWED_BCI_REPO_OS_VERSIONS,
+    reason="no included BCI repository - can't test",
```
Comment (alexandrevicenzi) [inline tests/test_repository.py:162]: > not included in the BCI repository

this is a skipif, there's no need to include "can't test", the logs will already produce a message like:

> SKIPPED: not included in the BCI repository

Comment (dirkmueller) [reply tests/test_repository.py]: that changes the meaning.. (it's not about it not being in the bci repo but about the bci repo not being in those containers)

Comment (alexandrevicenzi) [reply tests/test_repository.py:162]: still not clear to me, but ok.

Comment (dirkmueller) [reply tests/test_repository.py]: I've removed it everywhere now (it was just a copy of all the previous messages)

**Review by alexandrevicenzi — APPROVED** [review]

File: tests/test_repository.py [hunk 2]
```diff
@@ -159,7 +159,32 @@ def test_package_installation(container_per_test, pkg):
 
 @pytest.mark.skipif(
     OS_VERSION not in ALLOWED_BCI_REPO_OS_VERSIONS,
-    reason="no included BCI repository - can't test",
+    reason="no included BCI repository",
+)
```
Comment (dcermak) [inline tests/test_repository.py:163]: This needs a skipif for Tumbleweed too

**Review by dcermak — CHANGES_REQUESTED** [review]

File: tests/test_repository.py (hunk 2 above)
Comment (dirkmueller) [reply tests/test_repository.py]: also for 16.0. I made it 15.x specific.

File: tests/test_repository.py [hunk 3]
```diff
@@ -177,9 +177,37 @@ def test_repo_content_licensing(container_per_test) -> None:
     ), "Found a package with a SUSE-Firmware license"
 
 
+@pytest.mark.skipif(
+    (
+        OS_VERSION.startswith("15")
+        and OS_VERSION not in ALLOWED_BCI_REPO_OS_VERSIONS
+    ),
+    reason="no included BCI repository",
+)
```
Comment (dcermak) [inline tests/test_repository.py:186]: Are you sure this is the right condition? Because the test now runs on Tumbleweed

Comment (dirkmueller) [reply tests/test_repository.py]: now it is.

**Review by alexandrevicenzi — APPROVED** [review]

**Review by dcermak — APPROVED** [review]

**Pull request merged**